### DISCUSSION TITLE
"Crop Angle" command not working correctly - also changes Aspect

### DISCUSSION THREAD

**Discussion by scrollop** [rsjaffe/MIDI2LR#610] [Problems]
What should happen:

1) The "Crop Angle" command should function as per the "Angle" command under the crop tool ie. not change the Aspect or dimensions (unless the crop touches an image border).

What happens:

2) The "Crop Angle" command changes the "Angle" value however also changes the Aspect (dimensions).

Also, if we can have the ability to change the amount by which one click of a control wheel changes a value in LR (eg. at the moment the crop angle amount per click is slightly too high) that would be great.

Using the latest LR Classic CC on Win 10 and a Behringer X-touch compact.

**Comment by rsjaffe** [comment]
1. You can adjust the amount by which a click changes the angle by the app options dialog: https://github.com/rsjaffe/MIDI2LR/wiki/Plugin-Options-Dialog#limits . Adjust the straightenAngle limits: the narrower you make them, the finer the adjustments.
2. Could you check the beta release and see if that fixes the aspect problem? https://github.com/rsjaffe/MIDI2LR/releases/tag/v5.3.0.0

**Comment by scrollop** [comment]
Thanks for the advice.

1. Changing straightenAngle  doesn't change the "Crop angle" amounts. Perhaps it's a differnet tool/variable (eg. under Transform - rotate?)
2. The beta version doesn't fix the crop aspect changing bug.

Thanks.

**Comment by ghost** [comment]
I'm having the same issue, MIDI2LR version 5.3.0.2 and Lightroom Classic version 11.3.1.

I would also love a way to constraint the crop angle to smaller values than -45 to +45. The options currently doesn't work as @scrollop said (or it applies to Transform angle).

Thanks for the amazing work!

**Reply by rsjaffe** [reply]
Crop angle can be further constrained by adjusting the straighten angle limits: see https://github.com/rsjaffe/MIDI2LR/wiki/Plugin-Options-Dialog#other-settings

**Reply by ghost** [reply]
Doesn't work for me, it only applies to Transform angle, not Crop angle.

**Reply by ghost** [reply]
Oh I just realized there's "Crop Angle" AND "Straighten Angle" in the assignable commands. I thought the latter was the auto-straighten option, but it's actually a replacement of Crop Angle, and it doesn't change the aspect ratio!
I was poking around in the .lua files and I saw that in Database the "Crop Angle" option is noted as deprecated and replaced by "straightAngle". I think "Crop Angle" should maybe removed from the assignable commands altogether?

In any case, it's working perfectly as intended if I just pick Straighten Angle instead of Crop Angle.

**Reply by scrollop** [reply]
You're right! Awesome! Now if there was a way to reduce the amount it moves with one click...

**Reply by rsjaffe** [reply]
The more you constrain the limits, the less it moves with each click. I'm also going to remove Crop Angle. Deprecation isn't enough, as I can't expect people to always read the documentation, and I don't indicate it's deprecated when people open up the command menu.

**Reply by scrollop** [reply]
Thank you.

That's not a bad idea, though often need to straighten by 45 degrees or so (or more) so that wouldn't work.

Thanks for developing this- it saves hours of work! I donated this week.
